core/platform/FileUtils.cpp

```diff
@@ -715,48 +715,58 @@ std::string FileUtils::fullPathForDirectory(std::string_view dir) const
 {
     DECLARE_GUARD;
 
+    auto result = std::string();
+
     if (dir.empty())
     {
-        return "";
+        // result is ""
     }
-
-    if (isAbsolutePath(dir))
+    else if (isAbsolutePath(dir))
     {
-        return std::string{dir};
+        result = dir;
     }
-
-    // Already Cached ?
-    auto cacheIter = _fullPathCacheDir.find(dir);
-    if (cacheIter != _fullPathCacheDir.end())
+    else
     {
-        return cacheIter->second;
-    }
-    std::string longdir{dir};
-    std::string fullpath;
+        // Already Cached ?
+        auto cacheIter = _fullPathCacheDir.find(dir);
+        if (cacheIter != _fullPathCacheDir.end())
+        {
+            result = cacheIter->second;
+        }
+        else
+        {
+            std::string longdir{dir};
 
-    if (longdir[longdir.length() - 1] != '/')
-    {
-        longdir += "/";
-    }
+            if (longdir[longdir.length() - 1] != '/')
+            {
+                longdir += "/";
+            }
 
-    for (const auto& searchIt : _searchPathArray)
-    {
-        fullpath = this->getPathForDirectory(longdir, searchIt);
-        if (!fullpath.empty() && isDirectoryExistInternal(fullpath))
-        {
-            // Using the filename passed in as key.
-            _fullPathCacheDir.emplace(dir, fullpath);
-            return fullpath;
+            for (const auto& searchIt : _searchPathArray)
+            {
+                auto fullpath = this->getPathForDirectory(longdir, searchIt);
+                if (!fullpath.empty() && isDirectoryExistInternal(fullpath))
+                {
+                    // Using the filename passed in as key.
+                    _fullPathCacheDir.emplace(dir, fullpath);
+                    result = fullpath;
+                    break;
+                }
+            }
+
+            if (result.empty() and isPopupNotify())
+            {
+                AXLOG("axmol: fullPathForDirectory: No directory found at %s. Possible missing directory.", dir.data());
+            }
         }
     }
 
-    if (isPopupNotify())
+    if (not result.empty() and result.back() != '/')
```

Comment:
Please avoid using `not` and `and`, as they are not the convention in this project.

Reply:
Fixed